remc2/engine/CommandLineParser.h

```diff
@@ -48,47 +48,51 @@ class CommandLineParser {
         bool DoSetLevel() const {return m_set_level;};
         bool DoTestNetworkChng1() const {return m_test_network_chng1;};
 
+        // settings
+        std::string GetMemimagesPath() const {return m_memimages_path;};
+
     private:
         void InterpretParams();
 
         std::vector<std::string> m_params;
 
         // modes: these control behaviour and set multiple other parameters
-        bool m_mode_release_game {false};
-        bool m_mode_playing_game {false};
-        bool m_mode_test_regressions_game {false};
-        bool m_mode_debug_afterload {false};
-        bool m_mode_debug_onstart {false};
-        bool m_mode_test_network {false};
+        bool m_mode_release_game;
+        bool m_mode_playing_game;
+        bool m_mode_test_regressions_game;
+        bool m_mode_debug_afterload;
+        bool m_mode_debug_onstart;
+        bool m_mode_test_network;
 
         // parameters
-        bool m_alternative_gamespeed_control {true};
-        bool m_analyze_entity {true};
-        bool m_auto_change_res {false};
-        bool m_copy_skip_config {false};
-        bool m_debug_sequences {false};
-        bool m_debug_sequences2 {false};
-        bool m_detect_dword_a {false};
-        bool m_disable_graphics_enhance {false};
-        bool m_fix_flyasistant {false};
-        bool m_fix_mouse {false};
-        bool m_interval_save {false};
-        bool m_load_edited_level {false};
-        bool m_mouse_off {false};
-        bool m_mouse_off2 {false};
-        bool m_move_player {false};
-        bool m_no_show_new_procedures {false};
-        bool m_off_pause_5 {false};
-        bool m_right_button {false};
-        bool m_test_regression {false};
-        bool m_test_renderers {false};
-        bool m_debugafterload {false};
-        bool m_graphics_debug {false};
-        bool m_hide_graphics {false};
-        bool m_rotate_player {false};
-        bool m_set_objective {false};
-        bool m_set_level {false};
-        bool m_test_network_chng1 {false};
+        bool m_alternative_gamespeed_control;
```

Comment:
Why remove the initialization here. isn't  C and C++ known for letting junk in the memory effect initial values? Couldn't some of these default to true without this?

Reply:
I figured that it is redundant as I also set all the default values in `Init` which is also called from the constructor. I need the `Init` function, because this class is used as a global Variable and therefore the constructor cannot be used to pass `argc` and `argv`.
I thought about using a singleton pattern, but the additional indirection might actually be a performance problem when accessing a value deep in an inner loop.